content/200-concepts/100-components/01-prisma-schema/06-relations/420-relation-mode.mdx

```diff
@@ -0,0 +1,214 @@
+
+<!-- TODO: list actions with brief description? -->
+
+The following table shows which emulated referential actions are available for each database connector:
+
+| Database    | Cascade | Restrict | NoAction | SetNull | SetDefault |
+| :---------- | :------ | :------- | :------- | :------ | :--------- |
+| PostgreSQL  | **✔️**  | **✔️**   | **✔️**‡  | **✔️**  | **❌**†    |
+| MySQL       | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| SQLite      | **✔️**  | **✔️**   | **✔️**‡  | **✔️**  | **❌**†    |
+| SQL Server  | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| CockroachDB | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+| MongoDB     | **✔️**  | **✔️**   | **✔️**   | **✔️**  | **❌**†    |
+
+- † The `SetDefault` referential action is not supported.
+- ‡ The `NoAction` referential action is not supported for PostgreSQL and SQLite. Instead, use the `Restrict` action.
+
+Error messages returned by emulated referential actions in the `prisma` relation mode are generated by the Prisma Client and differ slightly from the error messages in the `foreignKeys` relation mode:
+
+```jsx
+Example:
+// foreignKeys:
+... Foreign key constraint failed on the field: `ProfileOneToOne_userId_fkey (index)`
+// prisma:
+... The change you are trying to make would violate the required relation 'ProfileOneToOneToUserOneToOne' between the `ProfileOneToOne` and `UserOneToOne` models.
+```
+
+### Introspection
+
+When you introspect a database with the `db pull` command with the `prisma` relation mode enabled, relations will not be automatically added to your schema. You will instead need to add any relations manually using the `@relation` attribute. This only needs to be done once – next time you introspect your database, Prisma will keep your added `@relation` attributes.
+
+### Prisma Migrate and <inlineCode>db push</inlineCode>
+
+When you apply changes to your Prisma schema with Prisma Migrate or `db push` with the `prisma` relation mode enabled, Prisma will not use foreign keys in your database.
+
+## Switch between relation modes
+
+It is only possible to switch between relation modes when you use a relational database connector (PostgreSQL, MySQL, SQLite, SQL Server, CockroachDB).
+
+### Switch from <inlineCode>foreignKeys</inlineCode> to <inlineCode>prisma</inlineCode>
+
+When you apply changes to your schema with Prisma Migrate or `db push`, if you switch the relation mode from `foreignKeys`to `prisma` Prisma will remove all previously created foreign keys in the next migration.
+
+If you keep the same database provider, you can then continue to work as normal. If you move from MySQL to PlanetScale, the migration files will contain foreign keys, which will return an error. We recommend that you delete the `migrations` directory and [use `db push` rather than Prisma Migrate](/guides/database/using-prisma-with-planetscale#differences-to-consider) if you use PlanetScale.
```

Comment:
"If you keep the same database provider," is confusing/misleading as this is both `provider = mysql`. PlanetScale _is_ MySQL.

I think what we want to say is that if you switch to a database that does not support foreign keys at all (like PlanetScale), instead of just doing the change for performance or other reasons, old migrations will not work any more. (And that we recommend `db push` on PlanetScale is more an aside)

So maybe:
```suggestion
If you switch to a database that does not support foreign keys at all, your existing migration history (which includes SQL DDL that creates foreign keys) might trigger errors if you ever have to start your database from nothing. We recommend that you delete the `migrations` directory (If you use PlanetScale we generally recommend you [use `db push` rather than Prisma Migrate](/guides/database/using-prisma-with-planetscale#differences-to-consider))
```
(english is terrible on that, but I think the facts are better)

Reply:
Agree that's better, have reworded